### PR TITLE
fix: call properly LoginForm cleanup method

### PR DESCRIPTION
**Pull request by yedpodtrzitko**
Hi,
I'm not really proficient with Javascript, so I'm open to suggestions for improvements. See my comments below for more details about the changes.

### CODE REVIEW

File: src/LoginForm.jsx [hunk 1]
```diff
@@ -185,7 +179,7 @@ class LoginForm extends React.Component {
           trySubscribe();
         }
       };
-      testUpdateLoop = setInterval(testConnectionUpdateLoop, 3000);
+      this.testUpdateLoop = setInterval(testConnectionUpdateLoop, 3000);
```
Comment (yedpodtrzitko) [inline src/LoginForm.jsx:182]: `_cleanup()` cant reference a local variable, so I assume this should be `this.testUpdateLoop` and the variable with identical name inside this method should not exist.

File: src/LoginForm.jsx [hunk 2]
```diff
@@ -107,7 +107,7 @@ class LoginForm extends React.Component {
   };
 
   componentWillUnmount = () => {
-    this._cleanup();
+    this._cleanup().then(() => console.log('cleanup complete'))
```
Comment (yedpodtrzitko) [inline src/LoginForm.jsx:110]: `_cleanup()` is declared as `async` method, so just calling it won't do anything. Also there's multiple `await` inside of it, so I'm not sure how to `await` all of them here.

File: src/LoginForm.jsx [hunk 3]
```diff
@@ -151,13 +151,7 @@ class LoginForm extends React.Component {
     this.setState({test: true})
     this._testStep('biz', 'pending');
     let client = this.props.createClient();
-    let testUpdateLoop = null;
 
-
-    window.onunload = () => {
-      cleanup()
```
Comment (yedpodtrzitko) [inline src/LoginForm.jsx:158]: `cleanup()` does not exist anywhere. I assume this should be `this._cleanup()`, but when I was trying to call it here, it did not execute at all. Maybe `onunload` is not allowed to execute async code?

**Comment by adwpc** [comment]
Thanks. Sorry for late, seems out of date.
Feel free to reopen if needed.

**Pull request closed without merging**